test/provider/AppWindow.unittest.ts

```diff
@@ -0,0 +1,164 @@
+import 'reflect-metadata';
+
+import {Identity} from 'openfin/_v2/main';
+
+import {createMockChannel} from '../mocks';
+import {Application} from '../../src/client/main';
+import {AbstractAppWindow} from '../../src/provider/model/AppWindow';
+import {ContextChannel} from '../../src/provider/model/ContextChannel';
+
+class TestAppWindow extends AbstractAppWindow {
+    private readonly _identity: Readonly<Identity>;
+
+    constructor(identity: Identity, appInfo: Application, channel: ContextChannel, creationTime: number | undefined, appWindowNumber: number) {
+        super(identity, appInfo, channel, creationTime, appWindowNumber);
+
+        this._identity = identity;
+    }
+
+    public get identity() {
+        return this._identity;
+    }
+
+    public bringToFront = jest.fn<Promise<void>, []>();
+    public focus = jest.fn<Promise<void>, []>();
+}
+
+const mockChannel = createMockChannel();
+
+const fakeIdentity = {uuid: 'test', name: 'test'};
+const fakeAppInfo: Application = {
+    appId: 'test',
+    name: 'Test App',
+    manifest: 'test.manifest',
+    manifestType: 'openfin'
+};
+
+beforeEach(() => {
+    jest.resetAllMocks();
+});
+
+describe('When querying if a window has a context listener', () => {
```

Comment:
As file is just simple generic `AppWindow` tests might aswell add in some intent tests aswell.